src/OrchardCore.Modules/OrchardCore.DataProtection.Azure/BlobModularTenantEvents.cs

```diff
@@ -0,0 +1,39 @@
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Environment.Shell.Removing;
+using OrchardCore.Modules;
+
+namespace OrchardCore.DataProtection.Azure;
+
+/// <summary>
+/// A tenant event that deletes data protection blobs from a container when a tenant is deleted.
+/// </summary>
+public class BlobModularTenantEvents : ModularTenantEvents
+{
+    private readonly BlobOptions _blobOptions;
+    private readonly ILogger<BlobModularTenantEvents> _logger;
+
+    public BlobModularTenantEvents(
+        BlobOptions blobOptions,
+        ILogger<BlobModularTenantEvents> logger)
+    {
+        _blobOptions = blobOptions;
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// Removes the data protection blob from the container when a tenant is deleted.
+    /// </summary>
+    /// <param name="context">The <see cref="ShellRemovingContext"/></param>
+    public async override Task RemovingAsync(ShellRemovingContext context)
+    {
+        var blobClient = new BlobClient(
```

Comment:
Why are you creating an instance of BlobClient here instead of using `IShellsFileStore`

Reply:
`IShellsFileStore` is about storing shell settings (i.e. what's normally in `App_Data\Sites\Default\appsettings.json` and similar), so that's not really applicable here. But `BlobFileStore` could be used instead indeed.

Reply:
yes any implementation of `IShellsFileStore` should be used instead. I think we should use the same instance used when creating the tenant.

Reply:
But `IShellsFileStore` is not for this as I elaborated above. It'll only have an instance if [Azure Shells Configuration Provider](https://docs.orchardcore.net/en/latest/reference/core/Shells/#azure-shells-configuration-provider) is used, but that feature is independent of [`OrchardCore.DataProtection.Azure`](https://docs.orchardcore.net/en/latest/reference/modules/DataProtection.Azure/) and connects to a different Azure Blob Storage container.

Reply:
But I think that will be a breaking change. The configuration used for the OrchardCore.DataProtection.Azure module and the for example the Media module that uses BlobFileStore differs. The file store configuration have a BasePath that I think is treated as a virtual folder when saving blobs. The DataProtection have a BlobName that points to a specific blob.

Here only a specific blob belonging to the deleted tenant should be deleted. It doesn't need the functionality from the more feature rich FileStore.

Reply:
Looking at the implementation of `OrchardCore.DataProtection.Azure`, it's fine to use `BlobClient` as you are doing for now. 

At some point, we should refactor the implementation to use `IDataProtectionFileStore` this would allow us to reuse the code in a better fashion.

Reply:
So as said, this can use `BlobFileStore`. That's the lower-level service which is suitable for any Blob Storage use case.

Reply:
yes but we also don't use it when we construct the service. I am okay with using `BlobFileStore` because its better. I am also okay with not using it too since we are already not using it.

Reply:
We can't use it under https://github.com/OrchardCMS/OrchardCore/blob/0d57489eca05d1924d6bd0eb56cbda170812c6f5/src/OrchardCore.Modules/OrchardCore.DataProtection.Azure/Startup.cs#L37 because `PersistKeysToAzureBlobStorage` requires a `BlobClient`. But it's not a big deal here either, actually, let's leave it.